### DISCUSSION TITLE
corejs.version:'3.15' not work with @babel/preset-env

### DISCUSSION THREAD

**Discussion by h1orz** [babel/babel#14973] [Q&A] [answered]
```
@babel/preset-env: 7.19.1
core-js: 3.25.2
```
I use `Array.prototype.groupBy` in my code, sometime `@babel/preset-env` not work, `@core-js/proposals/array-grouping` not import automatic. Webpack config like this:
```js
[
  '@babel/preset-env',
  {
    useBuiltIns: 'usage',
    corejs: {
      version: 3, // not work
      // version: '3.0', // not work
      // version: '3.15', // not work
      // version: '3.16', // work
      // version: '3.20', // work
      // version: '3.22', // work
      proposals: true
    },
    modules: false,
  },
]
```
When version <= ’3.15‘, `@babel/preset-env` can't import automatic. Is that some condition in there?

**Comment by liuxingbaoyu** [comment]
Do you mean >3.15 or <=3.15? This is to be expected as new support is continuously introduced as the version is updated.

**Reply by h1orz** [reply]
It is <=3.15, sorry. I corrected the description.

**Comment by h1orz** [comment, marked as answer]
https://github.com/zloirock/core-js/issues/1126
I got the answer, 'esnext.array.group-by' added in [`3.16`](https://github.com/zloirock/core-js/blob/ea7645f83b1df744c687bce222138ff68a42d0dc/packages/core-js-compat/src/modules-by-versions.mjs#L100).

**Reply by h1orz** [reply]
So, `@babel/preset-env` import automatic depend on `core-js-compat/modules-by-versions.json`? Whether or not actual version of `core-js` bigger then '3.15'?

**Reply by liuxingbaoyu** [reply]
I'm not quite sure I get what you mean. Babel will process it according to the passed `core-js` option.

**Reply by h1orz** [reply]
For example, I installed `core-js@3.25.2`, it is support `Array.prototype.groupBy`.
I installed `@babel/preset-env@7.19.1`, it is dependency `core-js-compat@^3.25.1`.
Then i set webpack config like this:
```js
[
  '@babel/preset-env',
  {
    useBuiltIns: 'usage',
    corejs: {
      version: '3.15', // or 3, but not 3.25
      proposals: true
    },
    modules: false,
  },
]
```
Run npm build, bundle.js do not included `Array.prototype.groupBy`'s polyfill.
If i set corejs.version '3.16', it well be included.
![image](https://user-images.githubusercontent.com/18734825/192014598-da44c56f-fb60-4dd2-a943-d9719195c6dd.png)
I think `@babel/preset-env` include condition of polyfill depend on [`core-js-compat/modules-by-versions.json`](https://github.com/zloirock/core-js/blob/master/packages/core-js-compat/src/modules-by-versions.mjs#L100).
`Array.prototype.groupBy` was added in `core-js@3.16`, so i set corejs.version >= '3.16' work well.
Even if `core-js@3.25.2` supported `Array.prototype.groupBy` , `@babel/preset-env` will still work on `corejs.version: '3.15'`'s relations.
Did I have an exact understanding?

**Reply by liuxingbaoyu** [reply]
Yes, what you understand is correct.
